alt1/scripts/notifications.ts

```diff
@@ -20,3 +29,109 @@ export function showToast(message: string, type: "error" | "success" = "success"
         setTimeout(() => toast.remove(), 500); // Remove from DOM
     }, 3000);
 }
+
+export function notifyEvent(event: EventRecord): void {
+    // early return if we aren't in alt1
+    if (!window.alt1) {
+        return;
+    }
+
+    const suppressToday = localStorage.getItem("toggleNotificationsToday") === "true";
+    if (suppressToday) {
+        return;
+    }
+
+    const notificationModes = JSON.parse(localStorage.getItem("notificationModes") ?? "[]");
+    const favoriteEventsRaw = localStorage.getItem("favoriteEvents");
+
+    // if favorite events are set, only show the favorites, otherwise, show all
+    if (favoriteEventsRaw) {
+        const favoriteEvents: string[] = JSON.parse(favoriteEventsRaw);
+        if (favoriteEvents.length > 0 && !favoriteEvents.includes(event.event)) {
+            return;
+        }
+    }
+
+    if (!notificationModes || notificationModes.length === 0) {
+        return;
+    }
+
+    const message = `${event.event} on w${event.world}`;
+
+    if (notificationModes.includes("tooltip")) {
+        showTooltip(message);
+    }
+
+    if (notificationModes.includes("toolbar")) {
+        showTitleBarText(event, `${message}`, event.duration * 1000);
+    }
+}
+
+export function showTooltip(message: string, time: number = 5000): void {
+    alt1.setTooltip(message);
+
+    if (tooltipTimeout) {
+        clearTimeout(tooltipTimeout);
+    }
+
+    tooltipTimeout = setTimeout(() => {
+        alt1.clearTooltip();
+    }, time);
+}
+
+export function showTitleBarText(event: EventRecord, message: string, duration: number = 120000): void {
```

Comment:
```suggestion
function showTitleBarText(event: EventRecord, message: string, duration: number = 120_000): void {
```
Just makes it clearer. Maybe specify that this is duration in ms. Not exported.